### ISSUE TITLE
ioc as a part of log record

### ISSUE THREAD

**Issue by kozeltom** [intfeed/sinkit-core#42]
Append data from ioc as a part of log record to make sorting (using feeds and taxonomies) over log records possible

**Comment by kozeltom** [comment]
enhanced log recrod:
```
{
    ...
    matched_iocs: [
        {
            id: "AU7_YAA99-sB3i2-8ttz" <string>
            ioc: {
                same as usual ioc, but following fields are omitted: virus_total_reports, seen.last, active 
            }
        },
        {
            ...
        }
    ]
}
```

**Comment by rattuscz** [comment]
Could you include some parts from virus_total_reports?
i use:
url_report->permalink
url_report->positives
url_report->total
to produce log record in dashboard. Also raw attribute is not needed IMHO.

**Comment by robcza** [comment]
Agree with raw, will omit that attribute. However I disagree with including VT report - this one is dynamic and could change a lot

**Comment by rattuscz** [comment]
Ok, permalink then at least? that should be.. permanent.

**Comment by robcza** [comment]
@kozeltom has to decide whether it is feasible to have the permalink there at the time when log is created. I think not.

**Comment by robcza** [comment]
```json
{
  "order": 0,
  "template": "logs*",
  "settings": {
    "index.refresh_interval": "5s",
    "index.number_of_shards": "1"
  },
  "mappings": {
    "match": {
      "numeric_detection": true,
      "dynamic_templates": [
        {
          "string_fields": {
            "mapping": {
              "index": "not_analyzed",
              "omit_norms": true,
              "type": "string"
            },
            "match_mapping_type": "string",
            "match": "*"
          }
        }
      ],
      "date_detection": true,
      "properties": {
        "reason": {
          "properties": {
            "fqdn": {
              "index": "not_analyzed",
              "type": "string"
            },
            "ip": {
              "index": "not_analyzed",
              "type": "string"
            }
          }
        },
        "request": {
          "properties": {
            "ip": {
              "index": "not_analyzed",
              "type": "string"
            },
            "raw": {
              "index": "not_analyzed",
              "type": "string"
            }
          }
        },
        "logged": {
          "format": "yyyy-MM-dd'T'HH:mm:ssZ",
          "type": "date"
        },
        "virus_total_request": {
          "properties": {
            "processed": {
              "format": "yyyy-MM-dd'T'HH:mm:ssZ",
              "type": "date"
            },
            "report_received": {
              "format": "yyyy-MM-dd'T'HH:mm:ssZ",
              "type": "date"
            },
            "status": {
              "index": "not_analyzed",
              "type": "string"
            }
          }
        },
        "matched_iocs": [
          {
            "properties": {
              "id": {
                "index": "not_analyzed",
                "type": "string"
              },
              "ioc": {
                "properties": {
                  "feed": {
                    "type": "object",
                    "properties": {
                      "name": {
                        "index": "not_analyzed",
                        "type": "string"
                      },
                      "url": {
                        "index": "not_analyzed",
                        "type": "string"
                      }
                    }
                  },
                  "protocol": {
                    "type": "object",
                    "properties": {
                      "application": {
                        "index": "not_analyzed",
                        "type": "string"
                      }
                    }
                  },
                  "description": {
                    "type": "object",
                    "properties": {
                      "text": {
                        "index": "not_analyzed",
                        "type": "string"
                      }
                    }
                  },
                  "source": {
                    "type": "object",
                    "properties": {
                      "fqdn": {
                        "index": "not_analyzed",
                        "type": "string"
                      },
                      "ip": {
                        "index": "not_analyzed",
                        "type": "string"
                      },
                      "bgp_prefix": {
                        "index": "not_analyzed",
                        "type": "string"
                      },
                      "reverse_domain_name": {
                        "index": "not_analyzed",
                        "type": "string"
                      },
                      "id": {
                        "type": "object",
                        "properties": {
                          "type": {
                            "index": "not_analyzed",
                            "type": "string"
                          },
                          "value": {
                            "index": "not_analyzed",
                            "type": "string"
                          }
                        }
                      },
                      "asn": {
                        "index": "not_analyzed",
                        "type": "string"
                      },
                      "url": {
                        "index": "not_analyzed",
                        "type": "string"
                      },
                      "asn_name": {
                        "index": "not_analyzed",
                        "type": "string"
                      },
                      "geolocation": {
                        "type": "object",
                        "properties": {
                          "cc": {
                            "index": "not_analyzed",
                            "type": "string"
                          },
                          "city": {
                            "index": "not_analyzed",
                            "type": "string"
                          },
                          "latitude": {
                            "type": "float"
                          },
                          "longitude": {
                            "type": "float"
                          }
                        }
                      }
                    }
                  },
                  "time": {
                    "type": "object",
                    "properties": {
                      "observation": {
                        "format": "yyyy-MM-dd'T'HH:mm:ssZ",
                        "type": "date"
                      },
                      "source": {
                        "format": "yyyy-MM-dd'T'HH:mm:ssZ",
                        "type": "date"
                      }
                    }
                  },
                  "classification": {
                    "type": "object",
                    "properties": {
                      "taxonomy": {
                        "index": "not_analyzed",
                        "type": "string"
                      },
                      "type": {
                        "index": "not_analyzed",
                        "type": "string"
                      }
                    }
                  },
                  "seen": {
                    "type": "object",
                    "properties": {
                      "first": {
                        "format": "yyyy-MM-dd'T'HH:mm:ssZ",
                        "type": "date"
                      }
                    }
                  }
                }
              }
            }
          }
        ],
        "action": {
          "index": "not_analyzed",
          "type": "string"
        },
        "client": {
          "index": "not_analyzed",
          "type": "string"
        }
      }
    },
    "aliases": {}
  },
  "aliases": {
    "logs": {}
  }
}
```

**Comment by kozeltom** [comment]
Permalink cannot be part of newly created log record, because it is known only after the virus total API is called. The permalink looks like: https://www.virustotal.com/url/befc8b3519a1b8a6e1f159e7b3124296f99fbc53512be8006e1ae1a244ce10a6/analysis/1439996441/
which contains some kind of hash and some number (id of analysis probably?) that we cannot guess, I would say.

**Comment by rattuscz** [comment]
ok, virustotal parts will be loaded separately from archive as it is now.
It's probably not needed for filtering ( can't think of use case ), only for display.

**Comment by robcza** [comment]
ok, this one seems solved
@kozeltom please close after merging
@rattuscz attributes from the index template posted above could be from now used for log filtering

**Comment by kozeltom** [comment]
Merged, closing.
